src/main/java/practice/StreamPractice.java

```diff
@@ -14,18 +18,28 @@
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(s -> Arrays.stream(s.split(",")))
+                .map(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
+                .min(Integer::compare)
+                .orElseThrow(() -> new NoSuchElementException("Can't get min value from list:"));
     }
-
     /**
      * Given a List of Integer numbers,
      * return the average of all odd numbers from the list or throw NoSuchElementException.
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
+
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
-    }
 
+        return IntStream.range(0, numbers.size())
+                .map(index -> index % 2 != 0 ? numbers.get(index) - 1 : numbers.get(index))
+                .filter(number -> (number % 2) != 0)
+                .mapToDouble(number -> number)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException("The list is empty!"));
```

Comment:
```suggestion
        return IntStream.range(0, numbers.size())
          .map(i -> i % 2 != 0 ? (numbers.get(i) - 1) : numbers.get(i))
          .filter(n -> n % 2 != 0)
          .average()
          .getAsDouble();
```

Yes, you are right Integers does't have average() operation, but IntStream is sequence of primitive int-valued elements, and `int's` have average().

Reply:
done